### PR TITLE
Fix prevent_sail_through workaround

### PR DESCRIPTION
**Pull request by sharcashmo**
Fix a workaround on PREVENT_SAIL_THROUGH by giving the ships to a new fleet

### CODE REVIEW

File: runorders.cpp [hunk 1]
```diff
@@ -2705,14 +2708,40 @@ int Game::DoGiveOrder(ARegion *r, Unit *u, GiveOrder *o)
...
+			// Check if fleets are compatible
+			if ((Globals->PREVENT_SAIL_THROUGH) &&
+					(!Globals->ALLOW_TRIVIAL_PORTAGE) &&
+					// flying ships are always transferrable
+					(ItemDefs[o->item].fly == 0)) {
+				// if target fleet had not sailed, just copy shore from source
+				if (fleet->prevdir == -1) {
+					fleet->prevdir = s->object->prevdir;
+				}
+
```
Comment (sgb) [inline runorders.cpp]: else?

**Comment by sharcashmo** [comment]
else NOP :D

Actually the following IF block could be moved into an else clause (*if
(fleet->prevdir != -1)* can be substituted by an *else*), if it's clearer.
The result would be the same.

The idea of that piece of code is that, if a ship from a restricted fleet
is moved into an unrestricted fleet in the same hex (it can happens when
creating a new fleet by giving all ships to a fresh unit, but also if
transferred to a fleet that has still not moved), the target fleet will get
the same restriction than the given ship had.

The following lines (the code inside *if (fleet->prevdir != -1)* clause)
just checks if fleets are compatible.

**Comment by sgb** [comment]
>
> else NOP :D
>
> Actually the following IF block could be moved into an else clause (*if
> (fleet->prevdir != -1)* can be substituted by an *else*), if it's clearer.
>

That is what what I meant :-)


> The result would be the same.
>

not entirely the same; the code in the second condition will (probably) get
executed as the body of the first condition will affect the condition of
the second.

**Comment by sharcashmo** [comment]
Yes, it will be executed, but the check will be successful as prevdir will
be the same for both fleets (target and source)

**Comment by sharcashmo** [comment]
It was just that I didn't want the first part (the copy of the prevdir) be
seen by a future developer as part of the check. For me they was
conceptually two different things: one the heritage of prevdir and another
one the check for compatibility.

But as you say maybe it is clearer the other way. As you prefer ;)

**Comment by artyomtrityak** [comment]
@sgb with `else` fixed, does everything else look good?

**Review by sgb — APPROVED** [review]
All looks good!

**Pull request merged**